src/bonsai/AccountTransactionSupervisor.ts

```diff
@@ -927,37 +916,323 @@ export class AccountTransactionSupervisor {
   }
 
   public async placeCompoundOrder(order: TradeFormPayload, source: TradeMetadataSource) {
-    if (order.orderPayload != null) {
+    const isMainOrderStateful =
+      order.orderPayload != null && !isShortTermOrderPayload(order.orderPayload);
```

Comment:
I stil prefer the `isStatefulOrder` from the client, but inverting shortTermOrder also works